mis_builder/models/mis_report_style.py

```diff
@@ -154,16 +155,49 @@ def merge(self, styles):
         return r
 
     @api.model
-    def render(self, lang, style_props, type, value):
+    def render(self, lang, style_props, type, value, currency=None):
         if type == 'num':
             return self.render_num(lang, value, style_props.divider,
                                    style_props.dp,
                                    style_props.prefix, style_props.suffix)
         elif type == 'pct':
             return self.render_pct(lang, value, style_props.dp)
+        elif type == 'mon':
+            return self.render_mon(lang, value, style_props.prefix,
+                                   style_props.suffix, currency)
         else:
             return self.render_str(lang, value)
 
+    @api.model
+    def render_mon(self, lang, value, prefix=None, suffix=None,
+                   currency=None):
+        # format number following user given or user currency
+        if value is None or value is AccountingNone:
+            return u''
+        if not currency:
+            currency = self.env.user.company_id.currency_id
+        elif isinstance(currency, int):
+            currency = self.env['res.currency'].browse(currency)
+        fmt = "%.{0}f".format(currency.decimal_places)
+        r = lang.format(
+            fmt, currency.round(value),
+            grouping=True, monetary=True).replace(r' ', u'\N{NO-BREAK SPACE}')
+        r = r.replace('-', u'\N{NON-BREAKING HYPHEN}')
+        if not prefix:
+            prefix = u''
+        if not suffix:
+            suffix = u''
+        if not prefix and currency.position == 'before':
+            prefix = u'{symbol}\N{NO-BREAK SPACE}'
+        elif not suffix:
+            suffix = u'\N{NO-BREAK SPACE}{symbol}'
```

Comment:
Why does prefix/suffix have precedence over the symbol?

Reply:
Maybe if you want to keep the number monetary formatting and rounding but have something else as currency symbol. by the way you can't replace the currency symbol by nothing.
For example 1000.04 if CHF would give 1'000.05 CHF and you may want 1'000.05 Swiss Francs.
Maybe we should give the possibility to show/hide currency symbol or choose between currency code / symbol.